### PR TITLE
[css-ui] Verify caret-color animations

### PR DESCRIPTION
**Pull request by mrego**
Added 2 new tests to check that "caret-color: auto" isn't interpolatible,
but "caret-color: currentcolor" is.

See w3c/csswg-drafts#781 for more information.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1162)

### CODE REVIEW

File: css-ui-3/caret-color-019.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+</style>
+<body>
+  <textarea id="textarea"></textarea>
+  <div id=log></div>
+
+  <script>
+    test(
+      function(){
+        var textarea = document.getElementById("textarea");
+        assert_equals(getComputedStyle(textarea).caretColor, 'rgb(255, 0, 0)');
```
Comment (frivoal) [inline css-ui-3/caret-color-019.html]: What is the point of this line and of `caret-color: red` higher up in the test? It seems to me that we could just drop them and still test what the assert claims this test is about.

File: css-ui-3/caret-color-020.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+</style>
+<body>
+  <textarea id="textarea"></textarea>
+  <div id=log></div>
+
+  <script>
+    test(
+      function(){
+        var textarea = document.getElementById("textarea");
+        assert_equals(getComputedStyle(textarea).caretColor, 'rgb(255, 0, 0)');
```
Comment (frivoal) [inline css-ui-3/caret-color-020.html]: What is the point of this line and of `caret-color: red` higher up in the test? It seems to me that we could just drop them and still test what the assert claims this test is about.

**Comment by frivoal** [comment]
Seems to me that both test contain a small unnecessary part. Other than that, they look good to me.

**Comment by mrego** [comment]
Yeah those lines were not needed, uploaded a new version without them.

BTW, this also depends on w3c/csswg-drafts#566 too (like #1161).

**Comment by frivoal** [comment]
> BTW, this also depends on w3c/csswg-drafts#566 too

Does it? CSS-UI says that [caret-color computes values the same was as the color property](https://drafts.csswg.org/css-ui-3/#propdef-caret-color). The color property says that [the computed value of color keywords is an rgb() value](https://drafts.csswg.org/css-color-3/#color0). css-animations says that [animations operate by changing the computed value](https://drafts.csswg.org/css-animations/#animations).

Both css-ui and css-color say that using fairly poor phrasing, as we're discussing in https://github.com/w3c/csswg-drafts/issues/741, but the intent is still unambiguous for this case.

It seems to me that the test is valid even before we decide that resolved values are used values for colors as argued in w3c/csswg-drafts#566. Am I missing something?

**Comment by mrego** [comment]
It's basically the first assert:
```js
        assert_equals(getComputedStyle(textarea).caretColor, 'rgb(255, 0, 255)');
```

It's checking the resolved style of `auto` (in 019 test) and `currentcolor` (in 020).
Without w3c/csswg-drafts#566 resolved this assert would be something like (in 019 test):
```js
        assert_equals(getComputedStyle(textarea).caretColor, 'auto');
```

**Comment by frivoal** [comment]
You're right, I missed that. I think we could (should) simplify the test and remove that dependency on w3c/csswg-drafts#566 by removing this:
```js
        player.currentTime = 0;
        assert_equals(getComputedStyle(textarea).caretColor, 'rgb(255, 0, 255)');
```
and (unrelated to the w3c/csswg-drafts#566, but still not necessary):
```js
        player.currentTime = 10;
        assert_equals(getComputedStyle(textarea).caretColor, 'rgb(0, 255, 0)');
```
from both tests, and only keeping `player.currentTime = 5` and the following assert, which is what really matters.

**Comment by mrego** [comment]
@frivoal good idea! I've removed the unneeded lines in the new version.

**Pull request merged**